core/src/test/java/org/apache/calcite/test/enumerable/EnumerableStringComparisonTest.java

```diff
@@ -0,0 +1,190 @@
+    return builder.getTypeFactory().builder()
+        .add(
+            "name",
+            builder.getTypeFactory().createTypeWithCharsetAndCollation(
+                builder.getTypeFactory().createSqlType(SqlTypeName.VARCHAR),
+                builder.getTypeFactory().getDefaultCharset(),
+                NO_CASE))
+        .build();
+  }
+
+  private RelDataType createVarcharNoCase(RelBuilder builder) {
+    return builder.getTypeFactory().createTypeWithCharsetAndCollation(
+        builder.getTypeFactory().createSqlType(SqlTypeName.VARCHAR),
+        builder.getTypeFactory().getDefaultCharset(),
+        NO_CASE);
+  }
+
+  @Test void testSortNoCase() {
+    tester()
+        .query("?")
+        .withRel(builder -> builder
+            .values(
+                createRecordVarcharNoCase(builder),
+                "Legal", "presales", "hr", "Administration", "MARKETING")
+            .sort(
+                builder.field(1, 0, "name"))
+            .build())
+        .explainHookMatches(""
+            + "EnumerableSort(sort0=[$0], dir0=[ASC])\n"
+            + "  EnumerableValues(tuples=[[{ 'Legal' }, { 'presales' }, { 'hr' }, { 'Administration' }, { 'MARKETING' }]])\n")
+        .returnsOrdered("name=Administration\n"
+            + "name=hr\n"
+            + "name=Legal\n"
+            + "name=MARKETING\n"
+            + "name=presales");
+  }
+
+  @Test void testMergeJoinOnStringNoCase() {
+    tester()
+        .query("?")
+        .withHook(Hook.PLANNER, (Consumer<RelOptPlanner>) planner -> {
+          planner.addRule(EnumerableRules.ENUMERABLE_MERGE_JOIN_RULE);
+          planner.removeRule(EnumerableRules.ENUMERABLE_JOIN_RULE);
+        })
+        .withRel(builder -> builder
+              .values(createRecordVarcharNoCase(builder),
+                  "Legal", "presales", "hr", "Administration", "MARKETING").as("v1")
+              .values(createRecordVarcharNoCase(builder),
+                  "Marketing", "bureaucracy", "Sales", "HR").as("v2")
+              .join(JoinRelType.INNER,
+                  builder.equals(
+                      builder.field(2, 0, "name"),
+                      builder.field(2, 1, "name")))
+              .project(
+                  builder.field("v1", "name"),
+                  builder.field("v2", "name"))
+              .build())
+        .explainHookMatches("" // It is important that we have MergeJoin in the plan
+            + "EnumerableMergeJoin(condition=[=($0, $1)], joinType=[inner])\n"
+            + "  EnumerableSort(sort0=[$0], dir0=[ASC])\n"
+            + "    EnumerableValues(tuples=[[{ 'Legal' }, { 'presales' }, { 'hr' }, { 'Administration' }, { 'MARKETING' }]])\n"
+            + "  EnumerableSort(sort0=[$0], dir0=[ASC])\n"
+            + "    EnumerableValues(tuples=[[{ 'Marketing' }, { 'bureaucracy' }, { 'Sales' }, { 'HR' }]])\n")
+        .returnsOrdered("name=hr; name0=HR\n"
+            + "name=MARKETING; name0=Marketing");
+  }
+
+  @Test void testStringComparison() {
```

Comment:
SqlOperatorBaseTest is based on Sql strings, this tests requires RelBuilder (since JavaCollation cannot be extracted from sql string), so for the moment it cannot be moved there.